lib/elixir/lib/enum.ex

```diff
@@ -899,14 +899,18 @@ defmodule Enum do
   end
 
   def find_index(enumerable, fun) when is_function(fun, 1) do
-    res =
-      Enumerable.reduce(enumerable, {:cont, 0}, fn(entry, acc) ->
-        if fun.(entry), do: {:halt, acc}, else: {:cont, acc + 1}
-      end)
+    res = Enumerable.reduce(enumerable, {:cont, {0, 0}}, fn(entry, acc) ->
+      {a, b} = acc
```

Comment:
We could pattern match this directly in the function head
I think the variable names could be a little clearer, and as we are not using the first one
we should prefix it with a `_`.